Exiled.Events/EventArgs/LockdowningEventArgs.cs

```diff
@@ -0,0 +1,54 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="LockdowningEventArgs.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.EventArgs
+{
+    using System;
+    using Exiled.API.Features;
+    using UnityEngine;
+
+    /// <summary>
+    /// Contains all informations before SCP-079 lockdowns a room.
+    /// </summary>
+    public class LockdowningEventArgs : EventArgs
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LockdowningEventArgs"/> class.
+        /// </summary>
+        /// <param name="player"><inheritdoc cref="Player"/></param>
+        /// <param name="roomGameObject"><inheritdoc cref="RoomGameObject"/></param>
+        /// <param name="auxiliaryPowerCost"><inheritdoc cref="AuxiliaryPowerCost"/></param>
+        /// <param name="isAllowed"><inheritdoc cref="IsAllowed"/></param>
+        public LockdowningEventArgs(Player player, GameObject roomGameObject, float auxiliaryPowerCost, bool isAllowed = true)
+        {
+            Player = player;
+            RoomGameObject = roomGameObject;
+            AuxiliaryPowerCost = auxiliaryPowerCost;
+            IsAllowed = isAllowed;
+        }
+
+        /// <summary>
+        /// Gets the player who's controlling SCP-079.
+        /// </summary>
+        public Player Player { get; }
+
+        /// <summary>
+        /// Gets the gameobject of the room that will be lockdowned.
+        /// </summary>
+        public GameObject RoomGameObject { get; }
```

Comment:
I think it would be better to make this also a setter (being able to lockdown completely different room, some dev would definetly use that) and getting an actual `Room` object instead of `GameObject`.
I'm still learning IL code, so I can't help you with that, but I think it would be possible.

Reply:
The only way to get room from gameobject as far as I know is Map.Rooms.First(room => room.gameobject = gameobject) although I don't think this is a good way of doing it

Reply:
I think that gameObject.GetComponentInParent<Room>() would also work, but I understand the problem that it may be inefficient.